From 0aaf27eff2e0e1d0d150df5546a13a9e7d8efa69 Mon Sep 17 00:00:00 2001
From: Simon1511 <me@simon1511.de>
Date: Fri, 5 Jan 2024 17:20:59 +0100
Subject: [PATCH 3/3] libsec-ril: Always return an empty value for
 GET_SMSC_ADDRESS

* Make parameter of type RilData (second last one) passed to SecRil::RequestComplete()
  in OnGetSmscAddressDone() be NULL, so that the framework passes an empty value when sending SMS
* Before U this value was not being used, on U the value returned is "SMSC",format_specifier
  which is considered invalid when passed back to RIL for sending SMS, however when an empty value
  is passed RIL will select the default SMSC, like it did on T and sending SMS works again

Change-Id: I608be406969d7fa58f15ef36340aff4ad1ce9fd0
Co-authored-by: Tim Zimmermann <tim@linux4.de>
---
 rw-system.sh | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/rw-system.sh b/rw-system.sh
index 95b7d06..3b373ec 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -335,6 +335,19 @@ changeKeylayout() {
     fi
 }
 
+if getprop ro.vendor.build.fingerprint | grep -qiE '^samsung/'; then
+    for f in /vendor/lib{,64}/libsec-ril.so; do
+        [ ! -f $f ] && continue
+        # shellcheck disable=SC2010
+        ctxt="$(ls -lZ "$f" | grep -oE 'u:object_r:[^:]*:s0')"
+        b="$(echo "$f" | tr / _)"
+
+        xxd -p -c0 "$f" | sed "s/600e40f9820c805224008052e10315aae30314aa/600e40f9820c805224008052e10315aa030080d2/g" | xxd -r -p > "/mnt/phh/$b"
+        chcon "$ctxt" "/mnt/phh/$b"
+        mount -o bind "/mnt/phh/$b" "$f"
+    done
+fi
+
 if [ "$(getprop ro.product.vendor.manufacturer)" = motorola ] && getprop ro.vendor.product.name |grep -qE '^lima';then
     for l in lib lib64;do
         for f in mt6771 lima;do
-- 
2.34.1

